### PR TITLE
[IMP][account_invoice_line_asset_category_required]Changed constraint by validation when is validated the invoice

### PR DESCRIPTION
**Pull request by luistorresm**
To solve: https://github.com/Vauxoo/addons-vauxoo/issues/775
- [x] Changed constraint on invoice line, by a validation when is validated the invoice.
- [x] Added tests in module

### CODE REVIEW

**Comment by luistorresm** [comment]
@dsabrinarg 

Can you check this PR please?

Regards

**Comment by luistorresm** [comment]
@JulioSerna 

May you give me +1?

File: account_invoice_line_asset_category_required/tests/test_invoice_line_asset_category.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self.category_asset = self.env.ref(
+            'account_asset.account_asset_category_fixedassets0')
+
+    def test_01_line_with_police_always_set(self):
+        "Create an invoice with a line that need category, ideal case"
+        invoice = self.invoice.copy()
+        for line in invoice.invoice_line:
+            line.account_id.user_type.write({'asset_policy': 'always'})
+            line.write({'asset_category_id': self.category_asset.id})
+        workflow.trg_validate(
```
Comment (JulioSerna) [inline account_invoice_line_asset_category_required/tests/test_invoice_line_asset_category.py]: @luistorresm 

Why not you use with the new api `signal_workflow` ?

**Comment by sabrinagnu** [comment]
@hbto May you please help us reviewing this development?, we already talk about this FIX before delegating the task, see if LGTY and can be merged.

Best regards!

**Comment by hbto** [comment]
@JulioSerna @luistorresm @dsabrinarg Looks good to me

+1

**Pull request merged**